### PR TITLE
Fixed CSR Authorizations.

### PR DESCRIPTION
**Pull request by sgilbride**
# Description

Fixed issue with CSR approval requiring the approval command to be used twice.
This issue was caused by a csr missing some of the required subjects. This led to an unhandled exception causing the csr approval to break after a cert was created. Since the cert was created before the unhandled exception, when an approval request was made again, the cert was detected, and the breaking logic was skipped over, allowing the csr approval to complete without issue on the second attempt.

Also updated some VC tests to match current jsonrpc functionality.

## Type of change

- [x ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested manually using dynamic agents.
Tested with pytests, especially web, auth, and control tests.

### CODE REVIEW

**Review by craig8 — COMMENTED** [review]
Please update documentation for create_csr as there is no name parameter.  Add documentation for the build_subject including return types.

**Comment by sgilbride** [comment]
@craig8 Updated docstrings for create_csr and build_subject

File: volttron/platform/certs.py [hunk 1]
```diff
@@ -503,8 +503,8 @@ def create_csr(self, fully_qualified_identity, remote_instance_name):
         The key that is used to sign the csr is <instance_name>.name.
 
         :param fully_qualified_identity:
-        :param target_volttron:
-        :return:
+        :param remote_instance_name:
+        :return csr.public_bytes:
```
Comment (craig8) [inline volttron/platform/certs.py:507]: So what exactly is csr.publich_bytes...https://cryptography.io/en/latest/hazmat/primitives/asymmetric/serialization/?highlight=public_bytes#cryptography.hazmat.primitives.serialization.PublicFormat

**Review by craig8 — CHANGES_REQUESTED** [review]

File: volttron/platform/certs.py (hunk 1 above)
Comment (sgilbride) [reply volttron/platform/certs.py]: It looks to be an encoded form of the certificate, which can be stored or transmitted. Updated docstring to better reflect this.

**Review by craig8 — APPROVED** [review]

**Review by shwethanidd — APPROVED** [review]

**Pull request merged**